### PR TITLE
Refactoring cluster plugin to a separate go module

### LINKED ISSUE #3431: Separate cluster plugin as a new go module
**Describe the feature request**
The cluster plugin should be made as a new git module

**Describe alternative(s) you've considered**
N/A

**Affected product area (please put an X in all that apply)**

- (X) APIs
- ( ) Addons
- (X) CLI
- ( ) Docs
- ( ) IAM
- ( ) Installation
- (X) Plugin
- ( ) Security
- (X) Test and Release
- (X) User Experience
- (X) Developer Experience

### PR DESCRIPTION
**Pull request by sathyanarays**
Signed-off-by: Sathyanarayanan Saravanamuthu <sathyanarays@vmware.com>

### What this PR does / why we need it
Cluster plugin needs to be on its own go module.

### Which issue(s) this PR fixes

Fixes #3431

### Describe testing done for PR
Tested by running Github pipelines

### Additional changes
- Modified github workflows to upload the coverage from tkg go modules as well

### Notes to reviewers
- The cluster plugin module still depends on the main tanzu-framework go module. The following are the packages:
    - github.com/vmware-tanzu/tanzu-framework/pkg/v1/tkr/controllers/source
    - github.com/vmware-tanzu/tanzu-framework/pkg/v1/tkr/pkg/utils
    - github.com/vmware-tanzu/tanzu-framework/pkg/v2/tkr/util/topology
- The above dependencies will go away when tkr is made as a independent go module as part of #3475

### CODE REVIEW

File: .github/workflows/main.yaml [hunk 1]
```diff
@@ -105,7 +105,7 @@ jobs:
     - name: Upload coverage reports to Codecov with GitHub Action
       uses: codecov/codecov-action@v3
       with:
-        files: ./coverage1.txt,./coverage2.txt,./coverage3.txt,./addons/coverage.txt,./pinniped-components/post-deploy/coverage.txt,./pinniped-components/tanzu-auth-controller-manager/coverage.txt
+        files: ./coverage1.txt,./coverage2.txt,./coverage3.txt,./addons/coverage.txt,./pinniped-components/post-deploy/coverage.txt,./pinniped-components/tanzu-auth-controller-manager/coverage.txt,./tkg/coverage.txt,./cmd/cli/plugin/cluster/coverage.txt
```
Comment (sathyanarays) [inline .github/workflows/main.yaml:108]: @Vandy-P , please add the management-cluster's coverage here!

File: Makefile [hunk 2]
```diff
@@ -499,6 +499,9 @@ test: generate manifests build-cli-mocks ## Run tests
 	# Test feature gates
 	$(MAKE) test -C featuregates
 
+	# Cluster plugin tests
+	$(MAKE) test -C cmd/cli/plugin/cluster
+
```
Comment (sathyanarays) [inline Makefile:504]: @Vandy-P , please add managementcluster tests here

File: cmd/cli/plugin/managementcluster/go.mod [hunk 3]
```diff
@@ -0,0 +1,243 @@
+module github.com/vmware-tanzu/tanzu-framework/plugin/managementcluster
+
+go 1.17
+
+replace (
+	github.com/vmware-tanzu/tanzu-framework => ../../../../
```
Comment (sathyanarays) [inline cmd/cli/plugin/managementcluster/go.mod:6]: @Vandy-P , can we remove unused replace statements here as well as in the cluster/go.mod?

Comment (sathyanarays) [reply cmd/cli/plugin/managementcluster/go.mod]: Done

File: Makefile (hunk 2 above)
Comment (Vandy-P) [reply Makefile:504]: Done

File: .github/workflows/main.yaml (hunk 1 above)
Comment (Vandy-P) [reply .github/workflows/main.yaml:108]: Done

**Review by tenczar — APPROVED** [review]
LGTM

File: cmd/cli/plugin/cluster/go.mod [hunk 4]
```diff
@@ -0,0 +1,245 @@
+module github.com/vmware-tanzu/tanzu-framework/plugin/cluster
+
+go 1.17
+
+replace (
+	github.com/vmware-tanzu/tanzu-framework => ../../../../
```
Comment (anujc25) [inline cmd/cli/plugin/cluster/go.mod:6]: I believe we will be able to remove this dependency after https://github.com/vmware-tanzu/tanzu-framework/pull/3475 gets in.

**Review by anujc25 — APPROVED** [review]
Changes looks good to me.

**Comment by sathyanarays** [comment]
//usebom:sandbox/v1.7.0-zshippable/427460516659595194/tkg-compatibility

File: .github/workflows/main.yaml [hunk 5]
```diff
@@ -105,8 +105,7 @@ jobs:
...
-
+        files: ./coverage1.txt,./coverage2.txt,./coverage3.txt,./addons/coverage.txt,./pinniped-components/post-deploy/coverage.txt,./pinniped-components/tanzu-auth-controller-manager/coverage.txt,./cli/core/coverage.txt,./cli/runtime/coverage.txt,./tkg/coverage.txt,./featuregates/client/cover.out,./featuregates/controller/cover.out,./capabilities/client/cover.out,./capabilities/controller/cover.out,./cmd/cli/plugin-admin/builder/cover.out,./cmd/cli/plugin-admin/codegen/cover.out,./cmd/cli/plugin-admin/test/cover.out,./apis/config/cover.out,./apis/run/cover.out,./packageclients/cover.out,./apis/addonconfigs/cover.out,./apis/cli/cover.out,./apis/core/cover.out,./cmd/cli/plugin/feature/cover.out,./cmd/cli/plugin/login/cover.out,./cmd/cli/plugin/package/cover.out,./cmd/cli/plugin/pinniped-auth/cover.out,./cmd/cli/plugin/secret/cover.out,./cmd/cli/plugin/cluster/coverage.txt,./cmd/cli/plugin/managementcluster/coverage.txt
```
Comment (marckhouzam) [inline .github/workflows/main.yaml:108]: @sathyanarays  Does the coverage files need to be added to `.github/workflows/release_staging.yaml` as well?

**Pull request merged**